compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -396,25 +396,24 @@ def process_pending_tasks() -> None:
                     )
                     res_spec: dict = json.loads(res_spec_s) or {}
 
+                    task_f = GCFuture(gc_task_id=tid)
+                    task_f.add_done_callback(forward_result)  # type: ignore[arg-type]
+
                     if fid and not self.function_allowed(fid):
                         # Same as web-service message but packed in a
                         # result error
-                        reject_msg = (
-                            f"Function {fid} not permitted on "
-                            f"endpoint {self.endpoint_id}"
-                        )
-                        log.warning(reject_msg)
+                        user_msg = "Function not permitted on endpoint"
+                        data = f"fid={fid}, eid={self.endpoint_id}"
+                        log.warning(f"{user_msg} [{data}]")
 
                         failed_result = Result(
                             task_id=tid,
-                            data=reject_msg,
+                            data=data,
```

Comment:
Here you changed the returned data from "Function xyz not permitted on endpoint abc" -> "fid=xyz, eid=abc" which is a lot less descriptive - from an obvious error string to echoing back the input arguments.

Yes the error is still in error_details but shouldn't the 'content' which is .data still highlight the fact that there is an issue?

Reply:
Thank you for pointing this out.  There's a larger conversation to be had here, and I don't want to tackle any of it at this juncture.  I've reverted the majority of this hunk now.